x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -125,29 +164,29 @@ export class ESSearchSource extends AbstractESSource {
     );
   }
 
-  async getFields() {
+  async getFields(): Promise<IField[]> {
     try {
       const indexPattern = await this.getIndexPattern();
-      return indexPattern.fields
-        .filter((field) => {
-          // Ensure fielddata is enabled for field.
-          // Search does not request _source
-          return field.aggregatable;
-        })
-        .map((field) => {
-          return this.createField({ fieldName: field.name });
-        });
+      const fields: IFieldType[] = indexPattern.fields.filter((field) => {
+        // Ensure fielddata is enabled for field.
+        // Search does not request _source
+        return field.aggregatable;
+      });
+
+      return (fields.map((field) => {
+        return this.createField({ fieldName: field.name }) as IField;
+      }) as unknown) as IField[];
     } catch (error) {
       // failed index-pattern retrieval will show up as error-message in the layer-toc-entry
       return [];
     }
   }
 
-  getFieldNames() {
-    return [this._descriptor.geoField];
+  getFieldNames(): string[] {
+    return typeof this._descriptor.geoField === 'string' ? [this._descriptor.geoField] : [];
```

Comment:
If geoField is not provided then an error is thrown in constructor so all of this check logic is not needed.